File: presentation/src/main/java/android/montreal/devfest/presentation/navigation/NavigatorWrapper.java
package android.montreal.devfest.presentation.navigation;

import android.content.Context;

class NavigatorWrapper implements Navigator.ViewMore, Navigator.Website {

    private final Context context;
    private final AppNavigator navigator;

    NavigatorWrapper(Context context, AppNavigator navigator) {
        this.context = context;
        this.navigator = navigator;
    }

    @Override
    public void showViewMore() {
        this.navigator.showViewMore(context);
    }

    @Override
    public void showWebsite() {
        this.navigator.showWebsite(context);
    }
}

File: stuff_presentation/src/main/java/android/montreal/devfest/stuff/presentation/navigation/NavigatorWrapper.kt
package android.montreal.devfest.stuff.presentation.navigation

import android.content.Context

internal class NavigatorWrapper(private val context: Context, private val navigator: AppNavigator) : Navigator.ViewMore, Navigator.Website {

    override fun showViewMore() {
        this.navigator.showViewMore(context)
    }

    override fun showWebsite() {
        this.navigator.showWebsite(context)
    }
}
